staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -185,48 +185,51 @@ func DefaultVerifyOptions() x509.VerifyOptions {
 
 // CommonNameUserConversion builds user info from a certificate chain using the subject's CommonName
 var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate) (*authenticator.Response, bool, error) {
-	if len(chain[0].Subject.CommonName) == 0 {
+	userInfo, succeeded := convertToUserInfo(chain[0].Subject)
+	if !succeeded {
 		return nil, false, nil
 	}
+	return &authenticator.Response{
+		User: userInfo,
+	}, true, nil
+})
 
-	isNewformat := false
-	tenant := ""
-	if len(chain[0].Subject.Organization) > 0 {
-		parts := strings.Split(chain[0].Subject.Organization[0], ":")
-		if len(parts) > 0 && parts[0] == "tenant" {
-			isNewformat = true
-			tenant = parts[1]
-		}
+func convertToUserInfo(subject pkix.Name) (*user.DefaultInfo, bool) {
+	if len(subject.CommonName) == 0 {
+		return nil, false
 	}
-
-	// backward compatible
-	// old format only uses /O
-	//if len(chain[0].Subject.OrganizationalUnit)==0 {
-	if !isNewformat {
-		parts := strings.Split(chain[0].Subject.CommonName, ":")
-		if len(parts) > 1 {
-			tenant = parts[0]
-		}
-		return &authenticator.Response{
-			User: &user.DefaultInfo{
-				Name:   chain[0].Subject.CommonName,
-				Groups: chain[0].Subject.Organization,
+	tenant := ""
+	if len(subject.Organization) > 0 {
+		parts := strings.Split(subject.Organization[0], ":")
+		if parts[0] == "tenant" {
```

Comment:
if O="tenant" (no ":" in the end), it should be identified as an old format. But the current logic will treat it as a new format. 

So I think line 204 should be len(parts) >1 && parts[0] == "tenant"

Reply:
I think O="tenant" should be treated as the new format.

Reply:
I see your point.